src/testdrive/src/action.rs

```diff
@@ -362,9 +475,45 @@ pub fn create_state(config: &Config) -> Result<State, Error> {
         kafka_admin_opts,
         kafka_producer,
         kafka_topics,
+        kinesis_client,
+        aws_region,
+        aws_account,
+        aws_access_key,
+        aws_secret_access_key,
+        token,
     })
 }
 
+fn get_kinesis_details_for_localstack() -> (
+    KinesisClient,
+    String,
+    String,
+    String,
+    String,
+    Option<String>,
+) {
+    let region = Region::Custom {
+        name: DUMMY_AWS_REGION.to_string(), // NB: This must match the region localstack is started up with!!
+        endpoint: LOCALSTACK_ENDPOINT.to_string(),
+    };
+    // Create a new KinesisClient
+    let provider = StaticProvider::new(
+        DUMMY_AWS_ACCESS_KEY.to_string(),
+        DUMMY_AWS_SECRET_ACCESS_KEY.to_string(),
+        None,
+        None,
+    );
+    let kinesis_client = KinesisClient::new_with(HttpClient::new().unwrap(), provider, region);
+    (
+        kinesis_client,
+        DUMMY_AWS_REGION.to_string(),
+        DUMMY_AWS_ACCOUNT.to_string(),
+        DUMMY_AWS_ACCESS_KEY.to_string(),
+        DUMMY_AWS_SECRET_ACCESS_KEY.to_string(),
+        None, // default to no token
+    )
+}
```

Comment:
I have some ideas on how to refactor this to hardcode less about localstack, but I don't want to hold up this PR on that.